pkg/controller/node/node_controller.go

```diff
@@ -113,15 +114,20 @@ func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result,
 		r.stosClient.clusterUID != cluster.GetUID() {
 
 		if err := r.setClientForCluster(cluster); err != nil {
-			log.Error(err, "failed to configure api client")
+			log.Error(err, "Failed to configure api client")
 			return reconcileResult, err
 		}
 	}
 
 	// Sync labels to StorageOS node object.
 	if err = r.syncLabels(instance.Name, instance.Labels); err != nil {
-		log.Error(err, "failed to sync labels, api may not be ready")
+		if storageoserror.ErrorKind(err) == storageoserror.APIUncontactable {
+			log.V(4).Info("Waiting for StorageOS API to become ready")
+			return reconcileResult, nil
+		}
+
 		// Error syncing labels - requeue the request.
+		log.Error(err, "Failed to sync node labels")
```

Comment:
I'm still seeing two repeated blocks of error in the logs when a cluster is created and node label sync is attempted.

One due to this error log line:
```
2019-07-08T07:18:57.128Z        ERROR   storageos.node  Failed to sync node labels      {"Request.Namespace": "", "Request.Name": "kind-control-plane", "error": "API error (Server failed to process your request. Was the data correct?): node not found"}
github.com/storageos/cluster-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error
        /go/src/github.com/storageos/cluster-operator/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/storageos/cluster-operator/pkg/controller/node.(*ReconcileNode).Reconcile
        /go/src/github.com/storageos/cluster-operator/pkg/controller/node/node_controller.go:130
github.com/storageos/cluster-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /go/src/github.com/storageos/cluster-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
github.com/storageos/cluster-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
        /go/src/github.com/storageos/cluster-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /go/src/github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/src/github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until
        /go/src/github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
```
And another one due to the returned error in the next line:
```
2019-07-08T07:18:57.128Z        ERROR   kubebuilder.controller  Reconciler error        {"controller": "node-controller", "request": "/kind-control-plane", "error": "API error (Server failed to process your request. Was the data correct?): node not found"}
github.com/storageos/cluster-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error
        /go/src/github.com/storageos/cluster-operator/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/storageos/cluster-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /go/src/github.com/storageos/cluster-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217
github.com/storageos/cluster-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
        /go/src/github.com/storageos/cluster-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /go/src/github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/src/github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until
        /go/src/github.com/storageos/cluster-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
```

Will it be better to convert these to info and not return error below. The following line re-queues the request already. Erroring may not be needed.

Reply:
Yes, I agree - it also fails when the node hasn't been created yet.  I'll convert to info and re-queue without returning error.